### PR TITLE
feat(dockerfile): Add checks for disabling signature checks for apk, apt-get, rpm, yum, dnf

### PR DESCRIPTION
**Pull request by james-otten-pan**
**By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.**

[//]: # "
    # PR Title
    Be aware that we use the title to create changelog automatically and therefore only allow specific prefixes
    - break:    to indicate a breaking change, this supersedes any of the types
    - feat:     to indicate new features or checks
    - fix:      to indicate a bugfix or handling of edge cases of existing checks
    - docs:     to indicate an update to our documentation
    - chore:    to indicate adjustments to workflow files or dependency updates
    - platform: to indicate a change needed for the platform
    Additionally a scope is needs to be added to the prefix, which indicates the targeted framework, in doubt choose 'general'.
    #    
    Allowed prefixs:
    ansible|argo|arm|azure|bicep|bitbucket|circleci|cloudformation|dockerfile|github|gha|gitlab|helm|kubernetes|kustomize|openapi|sca|secrets|serverless|terraform|general|graph|terraform_plan
    #
    ex.
    feat(terraform): add CKV_AWS_123 to ensure that VPC Endpoint Service is configured for Manual Acceptance
"

## Description

Add 4 new checks related to unsafe use of linux package managers in dockerfiles. Disabling these security features allows compromised packages to be installed, putting the image at risk.

## New/Edited policies
CKV2_DOCKER_7
CKV2_DOCKER_8
CKV2_DOCKER_9
CKV2_DOCKER_10
CKV2_DOCKER_11

### Description
CKV2_DOCKER_7
The `--allow-untrusted` option for `apk` allows packages with untrusted signature or no signature to be installed.
https://wiki.alpinelinux.org/wiki/Alpine_Package_Keeper

CKV2_DOCKER_8
The `--allow-unauthenticated` option for `apt-get` allow packages that cannot be authenticated to be installed.
https://manpages.ubuntu.com/manpages/jammy/man8/apt-get.8.html

CKV2_DOCKER_9
The `--nogpgcheck` option for `yum` and `dnf` allows packages without gpg signatures or with invalid gpg signatures to be installed.
https://man7.org/linux/man-pages/man8/dnf.8.html
https://man7.org/linux/man-pages/man8/yum.8.html

CKV2_DOCKER_10
The `--nodigest`, `--nosignature`, `--noverify`, `--nofiledigest` options for `rpm` disable signature checks.
https://man7.org/linux/man-pages/man8/rpm.8.html

CKV2_DOCKER_11
The `--force-yes` option for `apt-get` allow packages that cannot be authenticated to be installed, as well as packages to be downgraded, which could leave the system in a broken or inconsistent state.
https://manpages.ubuntu.com/manpages/jammy/man8/apt-get.8.html

### Fix
Avoid installing packages from untrusted sources and remove the unsafe options.

## Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] I have added tests that prove my feature, policy, or fix is effective and works
- [x] New and existing tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Review by gruebel — COMMENTED** [review]
as always, thanks for the contribution 💪 code-wise it looks great and always love the extensive test cases ❤️

File: checkov/dockerfile/checks/graph_checks/RunApkAllowUntrusted.yaml [hunk 1]
```diff
@@ -0,0 +1,11 @@
+metadata:
+  id: "CKV2_DOCKER_7"
+  name: "Ensure that packages with untrusted or missing signatures are not used"
```
[the remark this answers is not in the document]
Comment (james-otten-pan) [reply checkov/dockerfile/checks/graph_checks/RunApkAllowUntrusted.yaml]: Good call, changed the format.

File: checkov/dockerfile/checks/graph_checks/RunAptGetAllowUnauthenticated.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+metadata:
+  id: "CKV2_DOCKER_8"
+  name: "Ensure that packages with untrusted or missing signatures are not used by apt-get via the '--allow-unauthenticated' or '--force-yes' options"
```
Comment (james-otten-pan) [inline checkov/dockerfile/checks/graph_checks/RunAptGetAllowUnauthenticated.yaml:3]: Thoughts on splitting this into 2 like the ansible MR?

Comment (james-otten-pan) [reply checkov/dockerfile/checks/graph_checks/RunAptGetAllowUnauthenticated.yaml]: This change has been made

**Review by gruebel — APPROVED** [review]
🏆

**Pull request merged**